spotbugs/src/main/java/edu/umd/cs/findbugs/detect/ViewCFG.java

```diff
@@ -0,0 +1,176 @@
+                            " [shape=plaintext label=\" Handled exception for #" +
+                            edge.getSource().getExceptionThrower().getPosition() + "\"];");
+                    break;
+                case UNHANDLED_EXCEPTION_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Unhandled exception for #" +
+                            edge.getSource().getExceptionThrower().getPosition() + "\"];");
+                    break;
+                case RETURN_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Return\"];");
+                    break;
+                case START_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Start\"];");
+                    break;
+                case EXIT_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Exit" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Exit\"];");
+                    break;
+                case SWITCH_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Switch case (non-default)\"];");
+                    break;
+                case SWITCH_DEFAULT_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" Switch case (default)\"];");
+                    break;
+                case JSR_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" JSR statement\"];");
+                    break;
+                case RET_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" RET statement\"];");
+                    break;
+                case GOTO_EDGE:
+                    out.println("  Node" + src.getLabel() + " -> Node" + tgt.getLabel() +
+                            " [shape=plaintext label=\" GOTO statement\"];");
+                    break;
+                }
+            }
+            out.println("}");
+        } catch (IOException e) {
+            bugReporter.logError("Could not create file for method " + method.getName(), e);
+        }
+    }
+
+    private Path getMethodFile(Path classDir, String methodName) {
+        String methodFileName = methodName;
+        Path methodFile;
+        int index = 0;
+
+        do {
+            methodFile = Paths.get(classDir.toString(), methodFileName + ".dot");
+            methodFileName = methodName + ++index;
+        } while (Files.exists(methodFile));
```

Comment:
It is a simplification, yes, but the question is whether we want a second crypting string in the method name, since we already have one in the root directory name. I think it is less readable than just `foo.dot`, `foo1.dot` and `foo2.dot`. However, if you think it is better then I will change it.